### PR TITLE
Dataset validity 

### PR DESCRIPTION
**Pull request by rgottsc**
Copy dataset validity to layer validity on load & 
Calculate min/max over dataset validity and layer validities in view

### CODE REVIEW

**Comment by rgottsc** [comment]
[Dataset listing: Calculate min/max over dataset validity and layer validities in view](https://trello.com/c/vH4g7VNL/14-dataset-listing-calculate-min-max-over-dataset-validity-and-layer-validities-in-view)

**Comment by rgottsc** [comment]
[Dataset: Copy dataset validity to layer validity on load](https://trello.com/c/3eAB6693/13-dataset-copy-dataset-validity-to-layer-validity-on-load)

File: projects/wave-core/src/lib/users/user.service.ts [hunk 1]
```diff
@@ -428,7 +429,24 @@ export class UserService {
...
                                 );
-
+                                let globalDatasetMinVal = new TimePoint (source.time_start);
+                                let globalDatasetMaxVal = new TimePoint (source.time_end);
+                                sourceChannels.forEach((channelElement) => {
+                                    if (channelElement.time_start) {
+                                        globalDatasetMinVal = this.compareStartTime(channelElement.time_start, globalDatasetMinVal);
+                                    }
+                                    if (channelElement.time_end) {
+                                        globalDatasetMaxVal = this.compareStartTime(channelElement.time_end, globalDatasetMaxVal);
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/users/user.service.ts:439]:
```suggestion
                                        globalDatasetMaxVal = this.compareEndTime(channelElement.time_end, globalDatasetMaxVal);
```

File: projects/wave-core/src/lib/users/user.service.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ import {
 import {NotificationService} from '../notification.service';
 import {ColorizerData} from '../colors/colorizer-data.model';
 import {TimePoint} from '../time/time.model';
+import {element} from 'protractor';
```
Comment (ChristianBeilschmidt) [inline projects/wave-core/src/lib/users/user.service.ts:30]:
```suggestion
```

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**